org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CallHierarchyHandler.java

```diff
@@ -0,0 +1,337 @@
+		int character = params.getPosition().getCharacter();
+
+		try {
+			IMember candidate = getCallHierarchyElement(uri, line, character, monitor);
+			if (candidate == null) {
+				return null;
+			}
+			checkMonitor(monitor);
+			return Arrays.asList(toCallHierarchyItem(candidate));
+		} catch (OperationCanceledException e) {
+			// do nothing
+		} catch (JavaModelException e) {
+			JavaLanguageServerPlugin.log(e);
+		}
+
+		return null;
+	}
+
+	public List<CallHierarchyIncomingCall> callHierarchyIncomingCalls(CallHierarchyIncomingCallsParams params, IProgressMonitor monitor) {
+		Assert.isNotNull(params, "params");
+
+		CallHierarchyItem item = params.getItem();
+		Assert.isNotNull(item, "call item");
+		Position position = item.getSelectionRange().getStart();
+		int line = position.getLine();
+		int character = position.getCharacter();
+
+		try {
+			return getIncomingCallItemsAt(item.getUri(), line, character, monitor);
+		} catch (JavaModelException e) {
+			JavaLanguageServerPlugin.log(e);
+		} catch (OperationCanceledException e) {
+			// do nothing
+		}
+
+		return null;
+	}
+
+	public List<CallHierarchyOutgoingCall> callHierarchyOutgoingCalls(CallHierarchyOutgoingCallsParams params, IProgressMonitor monitor) {
+		Assert.isNotNull(params, "params");
+
+		CallHierarchyItem item = params.getItem();
+		Assert.isNotNull(item, "call item");
+		Position position = item.getSelectionRange().getStart();
+		int line = position.getLine();
+		int character = position.getCharacter();
+
+		try {
+			return getOutgoingCallItemsAt(item.getUri(), line, character, monitor);
+		} catch (JavaModelException e) {
+			JavaLanguageServerPlugin.log(e);
+		} catch (OperationCanceledException e) {
+			// do nothing
+		}
+
+		return null;
+	}
+
+	private IMember getCallHierarchyElement(String uri, int line, int character, IProgressMonitor monitor) throws JavaModelException {
+		Assert.isNotNull(uri, "uri");
+
+		ITypeRoot root = JDTUtils.resolveTypeRoot(uri);
+		if (root == null) {
+			return null;
+		}
+
+		if (root instanceof ICompilationUnit) {
+			ICompilationUnit unit = (ICompilationUnit) root;
+			if (root.getResource() == null) {
+				return null;
+			}
+			reconcile(unit, monitor);
+		}
+
+		checkMonitor(monitor);
+
+		// Try to locate the member at the given position.
+		IMember candidate = null;
+		int offset = JsonRpcHelpers.toOffset(root, line, character);
+		List<IJavaElement> selectedElements = codeResolve(root, offset);
+		Stream<IJavaElement> possibleElements = selectedElements.stream().filter(CallHierarchyCore::isPossibleInputElement);
+		Optional<IJavaElement> firstElement = possibleElements.findFirst();
+		if (firstElement.isPresent() && firstElement.get() instanceof IMember) {
+			candidate = (IMember) firstElement.get();
+		}
+
+		// If the member cannot be resolved, retrieve the enclosing method.
+		if (candidate == null) {
+			candidate = getEnclosingMember(root, offset);
+			if (candidate == null) {
```

Comment:
this whole if block is unnecessary. If Candidate is null, it'll be returned as such L158

Reply:
nice catch!